SEED-platform/seed
# ImportFile's post-matching mapping page shows historical mapping

#### Any background context you want to provide?
See issue linked below. The approach initially planned in the issue of creating a new ImportFile field to capture mapping settings during initial import was not needed as there was already a field capturing that info.

#### What's this PR do?
This PR uses ImportFile objects' `cached_mapped_columns` field to render the column mappings by user. 

**Very Notable Caveat**
The cached mappings makes use of raw, DB-level column names, so the current Column Display Name Settings are used to present cached mappings in a way they're accustomed to. For example, `PM Property ID` is shown instead of `pm_property_id`. The caveat here is that users can change Column Display Name Settings after the fact.

If this were to happen, technically, the value in SEED Header might not be _exactly_ what the user originally entered during import. **But** they will always get a reference to the DB-level column that the data file header was mapped to. 

The alternative is to provide the original text input as it was when the user imported the file.  This could be misleading or, worse, represent a completely different DB-level column that the data file header was mapped to on import.  

For example, if users change the display name of the DB column `custom_id_1` to `Some Unique ID` but they've previously imported files to `custom_id_1` when the display name was the default `Custom ID 1`. Users would need historical context to know that the data file header on these previously imported files are mapped to the column they currently understand as `Some Unique ID`.  

In this example, the current set up would show `Some Unique ID` as the display name instead of `Custom ID 1` that the user originally used on import.

#### How should this be manually tested?
Visit the mapping page of different ImportFiles and see the previously used mappings are used. This includes unit choices where applicable.

Also, import more files with the same columns as existed in previously imported files and map these data file columns to different SEED columns OR change column mappings for the Org. Then, verify that previously used mapping persist.

#### What are the relevant tickets?
#1992 

#### Screenshots (if appropriate)
